components/formats-gpl/src/loci/formats/in/VectraReader.java

```diff
@@ -0,0 +1,483 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the 
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public 
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package loci.formats.in;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import loci.common.Constants;
+import loci.common.DataTools;
+import loci.common.DateTools;
+import loci.common.RandomAccessInputStream;
+import loci.common.xml.XMLTools;
+import loci.formats.CoreMetadata;
+import loci.formats.FormatException;
+import loci.formats.FormatTools;
+import loci.formats.MetadataTools;
+import loci.formats.meta.MetadataStore;
+import loci.formats.tiff.IFD;
+import loci.formats.tiff.PhotoInterp;
+import loci.formats.tiff.TiffParser;
+import ome.xml.model.primitives.Color;
+import ome.xml.model.primitives.NonNegativeInteger;
+import ome.xml.model.primitives.Timestamp;
+
+import ome.units.quantity.Length;
+import ome.units.quantity.Time;
+import ome.units.UNITS;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * VectraReader is the reader for PerkinElmer Vectra QPTIFF data
+ */
+public class VectraReader extends BaseTiffReader {
+
+  // -- Constants --
+
+  /** Logger for this class. */
+  private static final Logger LOGGER =
+    LoggerFactory.getLogger(VectraReader.class);
+
+  /** TIFF image description prefix for PerkinElmer Vectra/QPTIFF files. */
+  private static final String SOFTWARE_CHECK = "PerkinElmer-QPI";
+
+  // -- Fields --
+
+  private int pyramidDepth = 1;
+  private String profileXML;
+
+  // -- Constructor --
+
+  /** Constructs a new Vectra reader. */
+  public VectraReader() {
+    super("PerkinElmer Vectra/QPTIFF", new String[] {"tiff", "tif", "qptiff"});
+    domains = new String[] {FormatTools.HISTOLOGY_DOMAIN, FormatTools.LM_DOMAIN};
+    noSubresolutions = true;
+    suffixSufficient = false;
+  }
+
+  // -- IFormatReader API methods --
+
+  /* (non-Javadoc)
+   * @see loci.formats.FormatReader#isThisType(java.lang.String, boolean)
+   */
+  @Override
+  public boolean isThisType(String name, boolean open) {
+    if (!open) {
+      return checkSuffix(name, "qptiff");
+    }
+    RandomAccessInputStream stream = null;
+    try {
```

Comment:
Channeling @mtbc here, this could probably be converted into a `try-with-resources`

Reply:
Addressed in https://github.com/openmicroscopy/bioformats/pull/3060